front-end/src/components/AnalyzersAndSeries.js

```diff
@@ -49,20 +49,47 @@ const AnalyzersAndSeries = ({ dimensions, columns, setDisplayPage, setShowModal,
 
         var selectColumns = [];
         for (let i = 0; i < dimensions; i++) {
-            selectColumns.push(JSON.parse(document.getElementsByClassName(i)[0].value));
+           
+            if (document.getElementsByClassName(i)[0].value === "") {
+                alert("Please select a column for the x-axis.");
+                return;
+            } else{
+                selectColumns.push(JSON.parse(document.getElementsByClassName(i)[0].value));
+            }
         }
-
+    
         seriesInfo.push({
             "columns": selectColumns,
             "analyze": {
                 "analysis": analyzers.filter(analyzer => document.getElementById(analyzer.name).checked)[0].code,
                 "analyzerValues": getAnalyzerOptions()
             }
-
         })
         console.log(seriesInfo)
     }
 
+    const handleSubmit = (e) => {
```

Comment:
Calling this "handleSubmit" when a handleSubmit is passed in as a prop causes an error